Modules/_testcapi/dict.c

```diff
@@ -331,6 +331,27 @@ dict_mergefromseq2(PyObject *self, PyObject *args)
 }
 
 
+static PyObject *
+dict_pop(PyObject *self, PyObject *args)
+{
+    PyObject *dict, *key;
+    if (!PyArg_ParseTuple(args, "OO", &dict, &key)) {
+        return NULL;
+    }
+    NULLABLE(dict);
+    NULLABLE(key);
+    PyObject *result = UNINITIALIZED_PTR;
+    int res = PyDict_Pop(dict, key,  &result);
+    if (res < 0) {
+        return NULL;
+    }
+    if (result == NULL) {
+        result = Py_NewRef(Py_None);
```

Comment:
```suggestion
        return PyLong_FromLong(res);
```

Reply:
It can be confused with actual None. For example `dict.get()` return an actual none, but several corresponding C API functions return NULL.

In other tests I made them returning AttributeError or KeyError, as it is less chance to confuse with real value, but I think that it would be better to use a special singleton `_testcapi.MISSING` in future.